### DISCUSSION TITLE
Build configurations

### DISCUSSION THREAD

**Discussion by thetic** [cpputest/cpputest#1712] [General]
The project currently contains configurations for several build systems and platform. Many of which are not exercised in CI. I've collected the ones I can find, but there may be more.

| Build Config | Appveyor | GitHub Actions | Travis (RIP) |
|- | - | - | - |
| [Autotools](https://github.com/cpputest/cpputest/blob/master/configure.ac) | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: 
| [CMake](https://github.com/cpputest/cpputest/blob/master/CMakeLists.txt) | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: 
| Makefile worker | | | :heavy_check_mark: 
| [msbuild](https://github.com/cpputest/cpputest/blob/master/tests/AllTests.vcxproj) | :heavy_check_mark: | | :heavy_check_mark: 
| [vcbuild](https://github.com/cpputest/cpputest/blob/master/tests/AllTests.vcproj) | :heavy_check_mark: | |
| [DOS Make](https://github.com/cpputest/cpputest/blob/master/platforms/Dos/Makefile) | | :heavy_check_mark: | :heavy_check_mark: 
| [Eclipse](https://github.com/cpputest/cpputest/tree/master/platforms/Eclipse-Cygwin) | | |
| [IAR](https://github.com/cpputest/cpputest/tree/master/platforms/iar)| | |
| [ARMCC Makefile](https://github.com/cpputest/cpputest/blob/master/platforms/armcc/Makefile) | | |
| [Symbian](https://github.com/cpputest/cpputest/tree/master/src/Platforms/Symbian) | | |
| [CCS](https://github.com/cpputest/cpputest/blob/master/platforms/CCStudio/CppUTest.pjt) | | |

I strongly suspect that the configs that aren't in CI are out of date. I think we should decide which of them we are willing and able to update and add to CI and I think we should delete the rest.

What do y'all think? Am I missing any? Are there any of these that we can easily get into CI or any that obviously can be delete?

**Comment by thetic** [comment]
It's woth noting that CMake can generate build systems for [Eclipse](https://cmake.org/cmake/help/latest/generator/Eclipse%20CDT4.html) and [Visual Studio back to version 6](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#id14). It works with ARMCC, IAR, OpenWatcom, and TI compilers among many more[1]. Removing explicit configuration for these tools would not mean dropping support.

[1]: https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html

**Comment by basvodde** [comment]
As long as a platform's only difference is the platform/name directory, then I'd rather keep them as they do not cause harm and are relatively easy to fix for the person who is picking that up again (although it is very unlikely Symbian will ever be picked up again :P)

**Reply by thetic** [reply]
I'm less concerned about the platforms than the build configs. I have no way to verify stuff like the IAR or CCS builds.

**Reply by basvodde** [reply]
Yes, I agree. Although we regularly get questions on these and every now and then a commit based on them. I've never used them myself but I'd say keep as long as we are suspecting use.

**Comment by basvodde** [comment]
The MakefileHelper build does seem to be missing from your list.

**Reply by thetic** [reply]
yup. i just started working on porting that to github actions

**Reply by basvodde** [reply]
Yup, that one ought to be supported as it is also used in @jwgrenning embedded TDD book.
